.github/workflows/go-test.yml

```diff
@@ -0,0 +1,51 @@
+name: Go tests
+on:
+  pull_request:
+    branches: [main]
```

Comment:
Currently, there is no need to run tests tests on tags or pushes to main, no ?

Reply:
Trying to add this in https://github.com/elastic/elastic-package/pull/1256

Reply:
I guess if those events are added then the `ref` in the `checkout` action needs to be updated depending on the event.

I found this that looks like could be also used here to get the right reference

https://github.com/elastic/apm-data/blob/a76e999543efb3ba803c9a57dd13a4f6b3ffa7e1/.github/workflows/bench-diff.yml#L21C1-L34

Reply:
That's true, I didn't recall that would not work for Windows.
Nice to be working with JS :muscle: !